### PR TITLE
Add impala adapter

### PR DESCRIPTION
**Pull request by PeteW**
This PR adds support for the apache impala database (a mysql-flavored database which runs on HDFS)

### CODE REVIEW

File: doc/dadbod.txt [hunk 1]
```diff
@@ -146,6 +146,11 @@ SQLite ~
     sqlite:/absolute/path
     sqlite:C:/windows/path
     sqlite:///relative/or/absolute/path
+
+                                                *dadbod-impala*
+Impala ~
+>
+    impala://[<host>[:<port>]]/[<database>]
```
Comment (tpope) [inline doc/dadbod.txt]: Move this out of the middle of the SQLite docs and into alphabetical order.

File: README.markdown [hunk 2]
```diff
@@ -17,6 +17,7 @@ take on [dbext.vim][], improving on it on the following ways:
   - Redis
   - SQL Server
   - SQLite
+  - Impala
```
Comment (tpope) [inline README.markdown]: Alphabetize.

**Comment by PeteW** [comment]
Understood. I didnt catch on to the alphabetized convention. Fixed thanks.
And Tim - thanks for all the stuff you write I end up using on a daily basis

**Comment by tpope** [comment]
Thanks for the contribution!

**Pull request merged**